tests/v2/validation/charts/gatekeeperallowednamespaces_test.go

```diff
@@ -0,0 +1,177 @@
+	yaml "gopkg.in/yaml.v3"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type GateKeeperAllowedNamespacesTestSuite struct {
+	suite.Suite
+	client                        *rancher.Client
+	session                       *session.Session
+	project                       *management.Project
+	gatekeeperChartInstallOptions *charts.InstallOptions
+}
+
+func (n *GateKeeperAllowedNamespacesTestSuite) TearDownSuite() {
+	n.session.Cleanup()
+}
+
+func (n *GateKeeperAllowedNamespacesTestSuite) SetupSuite() {
+	testSession := session.NewSession(n.T())
+	n.session = testSession
+
+	client, err := rancher.NewClient("", testSession)
+	require.NoError(n.T(), err)
+
+	n.client = client
+
+	// Get clusterName from config yaml
+	clusterName := client.RancherConfig.ClusterName
+	require.NotEmptyf(n.T(), clusterName, "Cluster name to install is not set")
+
+	// Get clusterID with clusterName
+	clusterID, err := clusters.GetClusterIDByName(client, clusterName)
+	require.NoError(n.T(), err)
+
+	// get latest version of gatekeeper chart
+	latestGatekeeperVersion, err := client.Catalog.GetLatestChartVersion(charts.RancherGatekeeperName)
+	require.NoError(n.T(), err)
+
+	// Create project
+	projectConfig := &management.Project{
+		ClusterID: clusterID,
+		Name:      gatekeeperProjectName,
+	}
+	createdProject, err := client.Management.Project.Create(projectConfig)
+	require.NoError(n.T(), err)
+	require.Equal(n.T(), createdProject.Name, gatekeeperProjectName)
+	n.project = createdProject
+
+	n.gatekeeperChartInstallOptions = &charts.InstallOptions{
+
+		ClusterName: clusterName,
+		ClusterID:   clusterID,
+		Version:     latestGatekeeperVersion,
+		ProjectID:   createdProject.ID,
+	}
+
+}
+func (n *GateKeeperAllowedNamespacesTestSuite) TestGateKeeperAllowedNamespaces() {
+	subSession := n.session.NewSession()
+	defer subSession.Cleanup()
+
+	client, err := n.client.WithSession(subSession)
+	require.NoError(n.T(), err)
+
+	if !client.Flags.GetValue(environmentflag.GatekeeperAllowedNamespaces) {
+		n.T().Skip()
+	}
+
+	n.T().Log("getting list of all namespaces")
+
+	sysNamespaces := settings.SystemNamespaces.Get()
+	sysNamespacesSlice := strings.Split(sysNamespaces, ",")
+	n.T().Log(sysNamespaces)
+
+	NSKinds := Kinds{
+		{ApiGroups: []string{""},
+			Kinds: []string{"Namespace"}},
+	}
+
+	NSMetadata := Metadata{
+		Name: "ns-must-be-allowed",
+	}
+
+	NSParameters := Parameters{
+		Namespaces: sysNamespacesSlice,
+	}
+
+	NSMatch := Match{
+		Kinds: NSKinds,
+	}
+
+	NSSpec := Spec{
+		EnforcementAction: "deny",
+		Match:             NSMatch,
+		Parameters:        NSParameters,
+	}
+
+	AllowedNamespaces := ConstraintYaml{
+		ApiVersion: "constraints.gatekeeper.sh/v1beta1",
+		Kind:       "K8sAllowedNamespaces",
+		Metadata:   NSMetadata,
+		Spec:       NSSpec,
+	}
+
+	yamlData, err := yaml.Marshal(&AllowedNamespaces)
+	require.NoError(n.T(), err)
+
+	fileName := "allowednamespaces.yaml"
+	err = os.WriteFile(fileName, yamlData, 0644)
```

Comment:
What's the need to write the yaml file? When you are reading it below on line 142

Reply:
this is now being done with a YAML string instead of creating a file